### PR TITLE
feat(*): migrate to algosdk v3.0.0

### PR DESCRIPTION
**Pull request by drichar**
## Description

This PR initiates the v4.0.0 release by migrating from `algosdk` v2 to v3. This is a breaking change that requires updates to transaction handling across all packages, establishing the foundation for additional v4 changes.

See the [official migration guide](https://github.com/algorand/js-algorand-sdk/blob/develop/v2_TO_v3_MIGRATION_GUIDE.md) for details about the underlying SDK changes.

⚠️ **Important**: Projects consuming this library will need to update their `algosdk` dependency to `^3.0.0` as it is a required peer dependency for compatibility with v4.

## Details
- Update `package.json` dependencies to `algosdk@3.0.0` across all packages
- Refactor transaction creation to use new type-based construction API
- Update transaction property access (`from` -> `sender`, `firstRound` -> `firstValid`)
- Modify transaction signing and verification logic for v3 compatibility
- Update test suites to use new transaction format and API
- Update all example projects to use new transaction parameter names

### CODE REVIEW

**Comment by drichar** [comment]
Reviews welcome! I'll leave this PR open for ~24-48 hours, which is hopefully enough time to get some eyeballs on it before merging.

Tagging a few contributors but anyone is welcome to submit a review.

@SilentRhetoric 
@PhearZero 
@acfunk 
@kieranroneill 
@gabrielkuettel 

After merging, I'll publish pre-release packages so developers can test v4's breaking changes in their projects.

Thanks! 🙏

File: packages/use-wallet/src/wallets/liquid.ts [hunk 1]
```diff
@@ -139,6 +139,7 @@ export class LiquidWallet extends BaseWallet {
       this.logger.debug('Signing transactions...', { txnGroup, indexesToSign })
 
       const authClient = this.authClient || (await this.initializeClient())
+      // @ts-expect-error - TODO: update liquid-auth-use-wallet-client to use algosdk v3
```
Comment (drichar) [inline packages/use-wallet/src/wallets/liquid.ts:142]: @HashMapsData2Value The `liquid-auth-use-wallet-client` package will need to support `algosdk` v3 in the next major release. The `Transaction` interfaces are incompatible between v2 + v3.

**Review by kieranroneill — APPROVED** [review]
This is my first look into v3 and I am liking it. From the Kibisis side, I didn't see any problems jump out, migration looks fine to me!

If there are any problems, I am more than happy to debug. I am giving an approval.

**Comment by acfunk** [comment]
As you know @drichar, I went through this process a few months ago [here](https://github.com/acfunk/use-wallet/tree/v4). These changes look pretty much the same, but more thorough. Great job! Looking forward to using `algosdk@3` in my apps as the responses are typed!

File: packages/use-wallet/src/wallets/lute.ts [hunk 2]
```diff
@@ -67,7 +67,10 @@ export class LuteWallet extends BaseWallet {
 
   private async getGenesisId(): Promise<string> {
     const algodClient = this.getAlgodClient()
-    const genesis = await algodClient.genesis().do()
+    const genesisStr = await algodClient.genesis().do()
```
Comment (SilentRhetoric) [inline packages/use-wallet/src/wallets/lute.ts:70]: This got way uglier--4 lines from 1?

**Review by SilentRhetoric — APPROVED** [review]

File: packages/use-wallet/src/wallets/lute.ts (hunk 2 above)
Comment (acfunk) [reply packages/use-wallet/src/wallets/lute.ts:70]: Yes, like everything else this method used to return `Promise<Record<string, any>>` which was parsable as JSON, but now is retuned as `Promise<string>` and so must be deserialized first.

Comment (drichar) [reply packages/use-wallet/src/wallets/lute.ts:70]: Unfortunately, yes 😅

**Comment by drichar** [comment]
The first beta release of v4.0.0 is now available! 🎉

You can find the release notes here: [v4.0.0-beta.1](https://github.com/TxnLab/use-wallet/releases/tag/v4.0.0-beta.1)

To test the new version in your project, upgrade using the "beta" tag, e.g.,

```bash
npm install @txnlab/use-wallet@beta
```

```bash
yarn upgrade @txnlab/use-wallet@beta
```

```bash
pnpm update @txnlab/use-wallet@beta
```

```bash
bun update @txnlab/use-wallet@beta
```

Obviously, make sure you have `algosdk` v3.0.0 installed as it's now a required peer dependency. Please try it out and let me know if you encounter any issues! 🙏

**Pull request merged**